frontend/src/contexts/AuthContext.tsx

```diff
@@ -1,10 +1,15 @@
-import { createContext, FC, useCallback, useContext, useState } from 'react'
+import { createContext, FC, useCallback, useContext } from 'react'
+import { useQueryClient } from 'react-query'
+
+import { LOGGED_IN_KEY } from '~constants/localStorage'
+import { useLocalStorage } from '~hooks/useLocalStorage'
+import * as AuthService from '~services/AuthService'
 
 type AuthContextProps = {
-  user?: any
   isAuthenticated?: boolean
-  login: () => void
-  logout: () => void
+  sendLoginOtp: typeof AuthService.sendLoginOtp
+  verifyLoginOtp: (params: { otp: string; email: string }) => Promise<void>
```

Comment:
do we really need to couple the auth _state_ (isAuthenticated, logout, login) with the auth _mechanism_ (OTP)? intuitively I'd think the auth context should be agnostic to auth mechanism, and just have the single responsibility of keeping track of the authentication state

Reply:
You make a great point. Will move the functions out

Reply:
Done